### PR TITLE
feat: Add Rust, web, Node.js example code

### PR DESCRIPTION
**Pull request by bsmth**
Checking in the finished examples to accompany https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_Wasm#rust_environment_setup

- build for the web
- target Bundlers (webpack in our case)

### CODE REVIEW

**Comment by MendyBerger** [comment]
@bsmth just wondering, if this example doesn't only work with node, why is it in a directory called rust-node? Why not rust-js?

**Comment by bsmth** [comment]
> Why not rust-js

Very good, I have no objections to that, shall I rename the directory?

File: rust-js/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Rust to WebAssembly in web and Node.js
+
+This directory contains examples of compiling Rust code to WebAssembly and using it on the web and in a Node.js application.
+The accompanying documentation can be found in the [Compiling from Rust to WebAssembly](https://developer.mozilla.org/en-US/docs/WebAssembly/Rust_to_Wasm) article on MDN.
+
+## Running the examples
+
+This document assumes you are following the instructions in the MDN article which covers prerequisites and how to install the tools you need to build the example.
```
Comment (dipikabh) [inline rust-js/README.md:8]:
```suggestion
This document assumes you're following the instructions in the above linked MDN article, which covers the prerequisites and the installation of the tools required to build the example.
```

File: rust-js/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+If you already have Rust, `wasm-pack` and Node.js installed and you would like to skip the setup steps, the following two sections have instructions on how to run examples that have already been compiled, focusing only on the differences between the two.
```
Comment (dipikabh) [inline rust-js/README.md:10]:
```suggestion
If you've already installed Rust, `wasm-pack`, and Node.js and wish to skip the setup steps, the next two sections provide instructions on how to run pre-compiled examples, focusing only on the differences between the two.
```
Not sure what "differences between the two" means. Should this be "the next two sections provide instructions on how to run pre-compiled examples, highlighting only the differences."?

File: rust-js/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+To run the example in your browser, you can open the `index.html` file:
+
+```bash
+# Target the web
+wasm-pack build --target web
+# Serve the page
+python3 -m http.server
+# or http-server via npm
+# http-server ./
```
Comment (dipikabh) [inline rust-js/README.md:22]: Should this be:
```suggestion
# or if http-server is installed via npm
http-server ./
```

File: rust-js/README.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+You should see an alert box that says "Hello, WebAssembly!".
+
+### Building and running an npm package
+
+To run the Node.js example, you will need to build the package targeting [Bundlers](https://rustwasm.github.io/docs/wasm-bindgen/reference/deployment.html#bundlers), install the Node.js dev dependencies and then serve the project:
```
Comment (dipikabh) [inline rust-js/README.md:29]:
```suggestion
To run the Node.js example, you will need to build the package targeting [Bundlers](https://rustwasm.github.io/docs/wasm-bindgen/reference/deployment.html#bundlers), install the Node.js dev dependencies, and then serve the project:
```

File: rust-js/README.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+```bash
+# Target Bundlers (webpack in our case)
+wasm-pack build --target bundler
+cd site
+# install the dev dependencies and run the server:
+npm i && npm run serve
+```
```
Comment (dipikabh) [inline rust-js/README.md:37]:
```suggestion
```bash
# Target bundlers (webpack in our case)
wasm-pack build --target bundler
cd site
# Install the dev dependencies and run the server
npm i && npm run serve
```
```

**Review by dipikabh — CHANGES_REQUESTED** [review]
I verified that the code snippets are the same as in content.
I've left a few edits for your consideration in the README file.

File: rust-js/README.md (hunk 2 above)
Comment (bsmth) [reply rust-js/README.md:10]: Good catch, updated your suggestion for clarity

**Review by dipikabh — APPROVED** [review]
This is great! Thanks for adding the example files here.

**Comment by bsmth** [comment]
Thank you!

**Pull request merged**